### PR TITLE
Project app: Localise project and project_pages

### PR DESCRIPTION
**Pull request by eatyourgreens**
Hook up `app_project` to the Panoptes `/translations` API for both the project and project page content. Workflows are already localised.
- Project translations are fetched after the project loads.
- Translated project fields are replaced by views, which are aliases for the corresponding translation strings.
- Project page translations are fetched in the page controller for each page, with the strings being added to each page as `page.strings`.
- `page.strings` is used to display the title and content of each page.

## Package
app-project

## How to Review
I've mocked a French and a Spanish locale for testing, so you should be able to browse translations, from Panoptes, for PH-TESS by adding the language code to the URL:
https://local.zooniverse.org:3000/projects/es/nora-dot-eisner/planet-hunters-tess?env=production
https://local.zooniverse.org:3000/projects/fr/nora-dot-eisner/planet-hunters-tess?env=production

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [x] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [x] Unit tests are included for the new feature
- [ ] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

File: packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js [hunk 1]
```diff
@@ -138,14 +144,17 @@ describe('Component > ProjectAboutPage & Connector', function () {
       fireEvent.click(dropdown)
       const navContainer = getByTestId('mobile-about-pages-nav')
       const links = getAllByRole(navContainer, 'link')
-      expect(links).to.have.lengthOf(3)
+      expect(links).to.have.lengthOf(2)
```
Comment (goplayoutside3) [inline packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js:147]: I don't think this test expects only 2 links. The mockStore provides a project with Research and Results pages. The Team page should also always be available in the browser, so the menu in this unit test should include 3 links (Research, Team, Results). Is something else going on here?

**Review by goplayoutside3 — COMMENTED** [review]
This is looking good! I can use the LocaleSwitcher on TESS to switch back and forth from English to French on the home page and about pages. More notes:
 
1. I don’t think `fr` dictionary files should be added with duplicate `en` static translations. When a `fr` static translation is missing, `i18next` will simply return the English translation values. The `locales/fr/` files should exist though, so until we’ve built a process to map PFE’s dictionaries to FEM’s in combination with Lokalise, let’s only add `fr/components.json` and `fr/screens.json` files containing an empty object `{}`.
    - Tangential note: In order to use LocaleSwitcher in response to languages available on a per-project basis, dictionary files for every language Zooniverse supports will need to exist in the codebase, and they can be blank files with an empty object `{}` for now. I think it makes sense to do this in a separate PR.
 
2. Now that translated project data are coming from `fetchTranslations()`, I think `fetchTranslations()` should have a spec file with unit tests in this PR. It’s a crucial data fetching function similar to `fetchProjectData()`.
 
3. In `getStaticPageProps.spec`, should a unit test be added to check the function is returning `strings`? I see a lot of unit tests for returned `workflows` data, but not for `project` data.
 
4. When someone tries to visit /fr/ for a project without french project translations, the app crashes instead of rendering the 404 page. I tried this on i-fancy-cats. I think there needs to be some error handling in `getStaticPageProps()` when `strings` does not exist.
 
5. I noticed ProjectAboutPage’s page title isn’t a static translation with `i18next` anymore. I think it makes most sense for the page title to always match the AboutSidebar navigation labels, and therefore be controlled as a static translation by the Zooniverse team. Perhaps project about pages’ titles should not be mutable in pandora and `strings = { content }`. What do you think?

File: packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js:147]: You're right. I saw two pages in the mock, so changed this, but there should be three in the links array. Something else is going on here.

**Comment by eatyourgreens** [comment]
>When someone tries to visit /fr/ for a project without french project translations, the app crashes instead of rendering the 404 page. I tried this on i-fancy-cats. I think there needs to be some error handling in getStaticPageProps() when strings does not exist.

After investigating a bit, Next only returns 404 if the locale is undefined. If the locale is defined, then the page is rendered. In that case, I'm adding default `strings` snapshots which should show the English language page eg. https://master.pfe-preview.zooniverse.org/projects/brooke/i-fancy-cats/?language=fr

**Comment by eatyourgreens** [comment]
Using the page title in a translation key is slightly dangerous, since the title is an editable field.
```js
const pageTitle = t(`About.PageHeading.title.${title.toLowerCase()}`)
```

This might be safer, since the URL key is constant.
```js
const pageTitle = t(`About.PageHeading.title.${page.url_key}`)
```

**Comment by eatyourgreens** [comment]
I think most of those points are addressed now:
- Document language is declared correctly on the `<html>` element.
- `fetchTranslations` has tests.
- The project and project pages set default values for strings after they load, so requests for translations that haven't been created yet won't error.
- About pages use static translations for the page headings, except in the document `<head>`.
- The test locales are empty objects.

File: packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js:147]: I Fancy Cats, which doesn't have a team page, works in the browser so something weird is going on with the tests.

Comment (eatyourgreens) [reply packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js:147]: I'd forgotten to change `page?.content` to `page?.strings?.content`. Should be fixed now.

**Comment by eatyourgreens** [comment]
I added some more tests this morning, so I think this is in much better shape now.

File: packages/app-project/pages/_document.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ export default class MyDocument extends Document {
 
   render () {
     return (
-      <Html lang="en">
+      <Html lang={this.props.locale}>
```
Comment (goplayoutside3) [inline packages/app-project/pages/_document.js:46]:
```suggestion
      <Html>
```

Next.js i18n-routing [automatically sets](https://nextjs.org/docs/advanced-features/i18n-routing#search-engine-optimization) the html `lang` tag, so I think we can remove any direct manipulation of it.

**Review by goplayoutside3 — APPROVED** [review]
Looks good! Just the note about NextJS and the html lang tag.

**Pull request merged**